lib/src/emoji_picker_internal_utils.dart

```diff
@@ -0,0 +1,108 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:emoji_picker_flutter/emoji_picker_flutter.dart';
+import 'package:flutter/services.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'emoji_lists.dart' as $emoji_list;
+
+/// Helper class that provides internal usage
+
+class EmojiPickerInternalUtils {
+  /// Establish communication with native
+  static const platform = MethodChannel('emoji_picker_flutter');
+
+  /// Restore locally cached emoji
+  Future<Map<String, String>?> _restoreFilteredEmojis(String title) async {
+    final prefs = await SharedPreferences.getInstance();
+    var emojiJson = prefs.getString(title);
+    if (emojiJson == null) {
+      return null;
+    }
+    var emojis =
+        Map<String, String>.from(jsonDecode(emojiJson) as Map<String, dynamic>);
+    return emojis;
+  }
+
+  // Get available emoji for given category title
+  Future<Map<String, String>> _getAvailableEmojis(Map<String, String> map,
+      {required String title}) async {
+    Map<String, String>? newMap;
+
+    // Get Emojis cached locally if available
+    newMap = await _restoreFilteredEmojis(title);
+
+    if (newMap == null) {
+      // Check if emoji is available on this platform
+      newMap = await _getPlatformAvailableEmoji(map);
+      // Save available Emojis to local storage for faster loading next time
+      if (newMap != null) {
+        await _cacheFilteredEmojis(title, newMap);
+      }
+    }
+
+    return newMap!;
```

Comment:
Not a big deal, just like to avoid `!`
```suggestion
    return newMap ?? {};
```